vendor/bat-native-ledger/src/bat/ledger/internal/endpoints/patch_wallets/patch_wallets.cc

```diff
@@ -0,0 +1,154 @@
+             std::string::npos) {
+    // HTTP 403
+    BLOG(0, "Request signature verification failure!");
+    return base::unexpected(Error::REQUEST_SIGNATURE_VERIFICATION_FAILURE);
+  } else {
+    BLOG(0, "Unknown message!");
+    return base::unexpected(Error::UNKNOWN_MESSAGE);
+  }
+}
+
+}  // namespace
+
+// static
+Result PatchWallets::ProcessResponse(const UrlResponse& response) {
+  switch (response.status_code) {
+    case net::HTTP_OK:  // HTTP 200
+      return {};
+    case net::HTTP_BAD_REQUEST:  // HTTP 400
+      BLOG(0, "Invalid request!");
+      return base::unexpected(Error::INVALID_REQUEST);
+    case net::HTTP_UNAUTHORIZED:  // HTTP 401
+      BLOG(0, "Bad request signature!");
+      return base::unexpected(Error::BAD_REQUEST_SIGNATURE);
+    case net::HTTP_FORBIDDEN:  // HTTP 403
+      return ParseBody(response.body);
+    case net::HTTP_CONFLICT:  // HTTP 409
+      BLOG(0, "geo_country already declared!");
+      return base::unexpected(Error::GEO_COUNTRY_ALREADY_DECLARED);
+    case net::HTTP_INTERNAL_SERVER_ERROR:  // HTTP 500
+      BLOG(0, "Unexpected error!");
+      return base::unexpected(Error::UNEXPECTED_ERROR);
+    default:
+      BLOG(0, "Unexpected status code! (HTTP " << response.status_code << ')');
+      return base::unexpected(Error::UNEXPECTED_STATUS_CODE);
+  }
+}
+
+PatchWallets::PatchWallets(LedgerImpl* ledger, std::string&& geo_country)
+    : RequestBuilder(ledger) {
+  if (geo_country.empty()) {
+    BLOG(0, "geo_country is empty!");
+    return;
+  }
+
+  base::Value::Dict content;
+  content.Set("geo_country", std::move(geo_country));
+
+  std::string json;
+  if (!base::JSONWriter::Write(content, &json)) {
+    BLOG(0, "Failed to write content to JSON!");
+    return;
+  }
+
+  content_ = json;
+}
+
+PatchWallets::~PatchWallets() = default;
+
+const char* PatchWallets::Path() const {
+  return "/v4/wallets/%s";
+}
+
+absl::optional<std::string> PatchWallets::Url() {
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(0, "Rewards wallet is null!");
+    return absl::nullopt;
+  }
+
+  DCHECK(!wallet->payment_id.empty());
+
+  return endpoint::promotion::GetServerUrl(
+      base::StringPrintf(Path(), wallet->payment_id.c_str()));
+}
+
+UrlMethod PatchWallets::Method() {
+  return UrlMethod::PATCH;
+}
+
+absl::optional<std::vector<std::string>> PatchWallets::Headers() {
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(0, "Rewards wallet is null!");
+    return absl::nullopt;
+  }
+
+  DCHECK(!wallet->payment_id.empty());
+  DCHECK(!wallet->recovery_seed.empty());
+
+  if (!content_) {
+    return absl::nullopt;
+  }
+
+  return util::BuildSignHeaders(
+      "patch " + base::StringPrintf(Path(), wallet->payment_id.c_str()),
+      *content_, wallet->payment_id, wallet->recovery_seed);
+}
+
+absl::optional<std::string> PatchWallets::Content() {
+  return std::move(content_);
```

Comment:
Makes sense, adding a comment there.

Reply:
Come to think of it, this is not a `RequestBuilder`-specific thing, as it's not `RequestBuilder` who relies on the ordering of calls to `Content()` and `Headers()`, but the endpoint implementations themselves.

Reply:
Actually, adjusting `RequestBuilder::Request()` a bit:
```
// ...

const auto content = Content();
if (!content) {
  return absl::nullopt;
}

auto headers = Headers(*content);
if (!headers) {
  return absl::nullopt;
}

// ...
```
so that endpoint implementations don't have to rely on the ordering of these two calls.

Reply:
This makes the ordering explicit rather than implicit, which is better (and self-documenting).